test/e2e/models/CodeServer.ts

```diff
@@ -49,46 +53,39 @@ export class CodeServer {
     // Make sure the editor actually loaded
     // If it's not visible after 5 seconds, something is wrong
     await this.page.waitForLoadState("networkidle")
-    return await this.page.isVisible("div.monaco-workbench", { timeout: 5000 })
+    return await this.page.isVisible(this.editorSelector)
   }
 
   /**
    * Focuses Integrated Terminal
-   * by going to the Application Menu
-   * and clicking View > Terminal
+   * by using "Terminal: Focus Terminal"
+   * from the Command Palette
+   *
+   * This should focus the terminal no matter
+   * if it already has focus and/or is or isn't
+   * visible already.
    */
   async focusTerminal() {
-    // If the terminal is already visible
-    // then we can focus it by hitting the keyboard shortcut
-    const isTerminalVisible = await this.page.isVisible("#terminal")
-    if (isTerminalVisible) {
-      await this.page.keyboard.press(`Control+Backquote`)
-      // Wait for terminal to receive focus
-      await this.page.waitForSelector("div.terminal.xterm.focus")
-      // Sometimes the terminal reloads
-      // which is why we wait for it twice
-      await this.page.waitForSelector("div.terminal.xterm.focus")
-      return
-    }
-    // Open using the manu
     // Click [aria-label="Application Menu"] div[role="none"]
     await this.page.click('[aria-label="Application Menu"] div[role="none"]')
 
     // Click text=View
     await this.page.hover("text=View")
     await this.page.click("text=View")
 
-    // Click text=Terminal
-    await this.page.hover("text=Terminal")
-    await this.page.click("text=Terminal")
+    // Click text=Command Palette
+    await this.page.hover("text=Command Palette")
+    await this.page.click("text=Command Palette")
+
+    // Type Terminal: Focus Terminal
+    await this.page.keyboard.type("Terminal: Focus Terminal")
+
+    // Click Terminal: Focus Terminal
+    await this.page.hover("text=Terminal: Focus Terminal")
+    await this.page.click("text=Terminal: Focus Terminal")
```

Comment:
This approach produces consistent results — no matter if the terminal is already focused, visible/not visible.